corehq/apps/case_importer/views.py

```diff
@@ -32,6 +33,14 @@ def render_error(request, domain, message):
     return HttpResponseRedirect(base.ImportCases.get_url(domain=domain))
 
 
+def validate_column_names(column_names, invalid_column_names):
```

Comment:
i was a [conversation](https://github.com/dimagi/commcare-hq/pull/20939#discussion_r196340691) with Norman over a PR regarding this.. Is there a case when it would be okay to actually update the original object by passing it as an argument instead of creating a new one and updating the original with each iteration?

Reply:
In most cases it's always better not to mutate input arguments since it makes it much harder to understand and reason about the code. If the rare case that you do decide to mutate an input argument the function shouldn't return anything to make it clear that there's some side effect taking place (which is what you've done here but not in the example from the other PR where you were returning something an mutating an arg).

One example where it makes sense to mutate an argument is in UCRs where one argument is the 'context' which can be used to do localized caching:
```python
def get_groups(user_id, context):
    groups = context.get_cache(key) or _get_groups(user_id)
    context.set_cache(key, groups)
    return groups
```

Reply:
Here's another example where we're building up a complex data structure: https://github.com/dimagi/commcare-hq/blob/6cc2bc81f1eb496912da07acd619b41daf3e4720/corehq/apps/app_manager/models.py#L1923